src/functions/assertions/HaveParameter.ps1

```diff
@@ -31,79 +31,50 @@
-            )
-            $i = $j = 0
-            do {
-                $token = $tokens[$i]
-                if ($token.Type -eq 'GroupStart') {
-                    $j++
-                }
-                if ($token.Type -eq 'GroupEnd') {
-                    $j--
-                }
-                if (-not $token.PSObject.Properties.Item('Depth')) {
-                    $token.PSObject.Properties.Add([Pester.Factory]::CreateNoteProperty("Depth", $j))
-                }
-                $token
-
-                $i++
-            } until ($j -eq 0 -or $i -ge $tokens.Count)
-        }
+        # Find param block
+        $ast = (& $SafeCommands['Get-Command'] -Name $Command).ScriptBlock.Ast
+
+        $paramBlock = $ast.FindAll(
+            {
+                param($Item)
+                return ($Item -is [System.Management.Automation.Language.ParamBlockAst])
+            },
+            $true
+        )
 
-        $errors = $null
-        $tokens = [System.Management.Automation.PSParser]::Tokenize($Command.Definition, [Ref]$errors)
+        foreach ($parameter in $paramBlock.Parameters) {
+            $paramInfo = & $SafeCommands['New-Object'] PSObject -Property @{
+                Name             = $parameter.Name.VariablePath.UserPath
+                DefaultValueType = $parameter.StaticType.Name
+                Type             = "[$($parameter.StaticType.Name.ToLower())]"
+            } | & $SafeCommands['Select-Object'] Name, Type, DefaultValue, DefaultValueType
 
-        # Find param block
-        $start = $tokens.IndexOf(($tokens | & $SafeCommands['Where-Object'] { $_.Content -eq 'param' } | & $SafeCommands['Select-Object'] -First 1)) + 1
-        $paramBlock = Get-TokenGroup $tokens[$start..($tokens.Count - 1)]
-
-        for ($i = 0; $i -lt $paramBlock.Count; $i++) {
-            $token = $paramBlock[$i]
-
-            if ($token.Depth -eq 1 -and $token.Type -eq 'Variable') {
-                $paramInfo = & $SafeCommands['New-Object'] PSObject -Property @{
-                    Name = $token.Content
-                } | & $SafeCommands['Select-Object'] Name, Type, DefaultValue, DefaultValueType
-
-                if ($paramBlock[$i + 1].Content -ne ',') {
-                    $value = $paramBlock[$i + 2]
-                    if ($value.Type -eq 'GroupStart') {
-                        $tokenGroup = Get-TokenGroup $paramBlock[($i + 2)..($paramBlock.Count - 1)]
-                        $paramInfo.DefaultValue = [String]::Join('', ($tokenGroup | & $SafeCommands['ForEach-Object'] { $_.Content }))
-                        $paramInfo.DefaultValueType = 'Expression'
-                    }
-                    else {
-                        $paramInfo.DefaultValue = $value.Content
-                        $paramInfo.DefaultValueType = $value.Type
-                    }
+            if ($null -ne $parameter.DefaultValue) {
+                if ([bool](Get-Member -Name Value -InputObject $parameter.DefaultValue -MemberType Property)) {
```

Comment:
This should also use `& $SafeCommands['Get-Member'] -Name ...`, but we can avoid the call completely with this. 🙂 
```suggestion
                if ($parameter.DefaultValue.psobject.properties['Value']) {
```